airbyte-cdk/python/unit_tests/sources/streams/test_streams_core.py

```diff
@@ -173,3 +175,11 @@ def test_wrapped_primary_key_various_argument(test_input, expected):
     wrapped = Stream._wrapped_primary_key(test_input)
 
     assert wrapped == expected
+
+
+@mock.patch("airbyte_cdk.sources.utils.schema_helpers.ResourceSchemaLoader.get_schema")
+def test_get_json_schema_is_cached(mocked_method):
+    stream = StreamStubFullRefresh()
+    for i in range(5):
+        stream.get_json_schema()
+    assert mocked_method.call_count == 1
```

Comment:
manually confirmed this test fails if [@lru_cache](https://github.com/airbytehq/airbyte/pull/18572/files#diff-49f3635a2319e7b734cc81dd557f7d14fa8b7501f981a46855bc9cee154607a0R113) is deleted